tests/datalab/test_datalab.py

```diff
@@ -981,53 +981,148 @@ def test_pred_probs_precedence(self, pred_probs, random_embeddings):
         )
 
 
-def make_data(num_examples=200, num_features=3, noise=0.2, error_frac=0.1, error_noise=5):
-    np.random.seed(SEED)
-    X = np.random.random(size=(num_examples, num_features))
-    coefficients = np.random.uniform(-1, 1, size=num_features)
-    label_noise = np.random.normal(scale=noise, size=num_examples)
-
-    true_y = np.dot(X, coefficients)
-    y = np.dot(X, coefficients) + label_noise
-
-    # add extra noisy examples
-    num_errors = int(num_examples * error_frac)
-    extra_noise = np.random.normal(scale=error_noise, size=num_errors)
-    random_idx = np.random.choice(num_examples, num_errors)
-    y[random_idx] += extra_noise
-    error_idx = np.argsort(abs(y - true_y))[-num_errors:]  # get the noisiest examples idx
-
-    # create test set
-    X_test = np.random.random(size=(num_examples, num_features))
-    label_noise = np.random.normal(scale=noise, size=num_examples)
-    y_test = np.dot(X_test, coefficients) + label_noise
-
-    return {
-        "X": X,
-        "y": y,
-        "true_y": true_y,
-        "X_test": X_test,
-        "y_test": y_test,
-        "error_idx": error_idx,
-    }
+class TestDatalabForRegression:
+    @pytest.fixture
+    def regression_data(
+        self, num_examples=200, num_features=3, noise=0.01, error_frac=0.05, error_noise=10
+    ):
+        np.random.seed(SEED)
+        X = np.random.random(size=(num_examples, num_features))
+        coefficients = np.random.uniform(-1, 1, size=num_features)
+        label_noise = np.random.normal(scale=noise, size=num_examples)
+
+        true_y = np.dot(X, coefficients)
+        y = np.dot(X, coefficients) + label_noise
+
+        # add extra noisy examples
+        num_errors = int(num_examples * error_frac)
+        extra_noise = np.random.normal(scale=error_noise, size=num_errors)
```

Comment:
```suggestion
        extra_noise = np.random.normal(scale=error_noise, size=num_errors) * max(1, error_noise)
```

Reply:
idea here is the test will be more stable if the extra_noise is more obvious. I don't think there's any reason to make this test challenging, it should be obvious errors IMO

Reply:
in future tests like this, can also shrink size of `num_errors` so that it's not affecting the ML model too much during training